client/mint_session/src/main.rs

```diff
@@ -27,18 +27,15 @@ pub extern "C" fn call() {
 
     let token_owner = runtime::get_named_arg::<Key>(ARG_TOKEN_OWNER);
     let token_metadata: String = runtime::get_named_arg(ARG_TOKEN_META_DATA);
-    let token_uri: String = runtime::get_named_arg(ARG_TOKEN_URI);
 
-    let (owned_tokens_dictionary_key, collection_name) = runtime::call_contract::<(Key, String)>(
+    let (collection_name, owned_tokens_dictionary_key, ) = runtime::call_contract::<(String, Key)>(
```

Comment:
this isn't the collection_name, it's the receipt_name

(same comment on other similar usages of mint and transfer elsewhere in the code)